### DISCUSSION TITLE
Updating Icon Size and Style in OpenFEDEM UI

### DISCUSSION THREAD

**Discussion by oeysstra** [openfedem/fedem-gui#10] [General]
## Updating Icon Size and Style in OpenFEDEM UI

The current icon size in the OpenFEDEM UI is 16x16 pixels. While this size was adequate in the past, it has become clear that larger and more modern icons are needed to improve usability and aesthetics. This discussion aims to come to a conclusion on the following aspects:

1. **Icon Design:**
   - How should the new icons look? (Style, color scheme, etc.)

2. **Icon Type:**
   - Should the icons be vector-based (scalable, e.g., SVG) or pixel-based (fixed size, e.g., PNG)?

3. **File Format:**
   - What file format should we use for the new icons?

Additionally, we seek input on:

- **Sources for Free Icons:**
  - Recommendations for sources where we can find free icons that are suitable for our project.

- **Tools for Conversion:**
  - Tools and methods to convert existing icons to the new size and format.

We welcome all suggestions, insights, and resources that the community can provide to help us make informed decisions and implement these changes effectively.

**Comment by kmokstad** [comment]
Don't know why this discussion was closed before anyone could comment on it (by accident?).
So I take the liberty to reopen it as I have some info on the matter:

I think using SVG-files for icons should be explored. I did some testing, and it turns out to be very little code changes required to add support for this. It even works with Qt 4.8! We need to include one extra DLL-fil with the installation though (QtSvg4.dll). In the image below, I have replaced the icons for Open and Save with two SVG-files that I found on the internet.

![image](https://github.com/openfedem/fedem-gui/assets/12747326/8e072fef-48ec-4394-8417-e276ebdbddf8)

There are many standard icons on SVG-format available out there, e.g., https://www.svgrepo.com/ where I found those two. Seems there are ways to edit them also. If there is a simple way of converting our old XPM-icons to SVG with improved resolution I don't know.

**Reply by oeysstra** [reply]
I was moving some discussion to different repos and by accident it could have been closed. Sorry for the inconvenience.

I agree with the @kmokstad that we should explore the SVG-files. I have stared som exploratory work using the IncScape tool, inkscape.org. For sure we can and should use standard icons, already made, for what is standard but I presume that we need some tailor made icons as well. 

I have made a python script that reads inn the 16x16 xpm icon files and converts them to png. I have tried various upscaling techniques, but there are no logics in those methods encoding straight lines, curves etc. @kmokstad, should we create a separate repo for this work?

**Reply by kmokstad** [reply]
I'm not sure we need a separate repository for the icons, as it then would be required to be included as yet another submodule of this repository. But anyone that would like to contribute making new icons are of course free to organize their work in separate repositories if convenient, but when getting to the point where one wants to include them in the GUI code, I rather see a pull request with the new icons files here.

I have open a new PR (#12) which includes the support for svg-icons. The last commit there (https://github.com/openfedem/fedem-gui/pull/12/commits/8d3af059389a029b4fa519fcc70556d7bbc30323) shows what little code changes are needed to replace the open and save icons with the respective svg-files. Those two icons are just added for the test - they are not supposed to be the final ones. Using a resource file for this purpose is nice, as the icon files then are "compiled" into the binary and do not need to be embedded with the installation itself, which may be a security issue.

One minor challenge is that most of the mechanism object icons (Triads, Joints, Springs, etc.) now are stored within the fedem-mdb repository. Therefore, we probably need to have a separate handling for those icons in the vpmDB library. Think that is solvable (saw a post on a similar matter earlier), but we can start replacing the standard icons now stored in this repository.

**Reply by kmoks** [reply]
On the last point above; the Icons currently stored in the vpmDB library are there because they also are used as pixmaps on the object tree of the model manager panel in the GUI, in addition to (for most of them) as icons on menu- and toolbar commands. Replacing the xpm-icons by svg or equivalent in the object three will require a bigger code change as the pixmaps here passed around as char** pointers in the class hierarchies. Therefore, I propose to leave the object tree icons as is for now, and only focus on the menu- and toolbar icons. Then we don't need to change anything in fedem-mdb for this issue.
